source/Mission.cpp

```diff
@@ -151,6 +151,20 @@ void Mission::Load(const DataNode &node)
 			blocked = child.Token(1);
 		else if(child.Token(0) == "deadline" && child.Size() >= 4)
 			deadline = Date(child.Value(1), child.Value(2), child.Value(3));
+		else if(child.Size() == 2 && child.Token(0) == "complete" && child.Token(1) == "anywhere")
+			completeAnywhere = true;
+		else if(child.Size() == 2 && child.Token(0) == "complete" && child.Token(1) == "at")
+		{
+			LocationFilter loaded(child);
+			if(!loaded.IsValid())
+				continue;
+			if(loaded.IsEmpty())
+				child.PrintTrace("Error: The \"complete at\" filter must not be empty. Ignoring this line.");
+			else
+				completionFilter = loaded;
+		}
```

Comment:
Checking the validity of these filters here probably isn't going to go well.
If they refer to any planets, systems, or outfits that haven't been loaded yet (because, at this point, loading is not complete,) those filters will appear invalid even if they will be valid once everything is fully loaded.

Reply:
I moved the validity check to Mission::IsValid()